### PR TITLE
Use Autoconf checks vs. OPENSSL_VERSION_NUMBER

### PR DESCRIPTION
**Pull request by jablko**
This will work better with the various OpenSSL forks.

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1314/ for details.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/1208/ for details.

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1316/ for details.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/1210/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1317/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1211/ for details.

File: iocore/net/BIO_fastopen.cc [hunk 1]
```diff
@@ -116,39 +108,18 @@ static long
...
-    return 0;
-
   case BIO_C_SET_CONNECT:
     // We only support BIO_set_conn_address(), which sets a sockaddr.
     ink_assert(larg == 2);
-    bio->ptr = ptr;
-    return 0;
-
-  // We are unbuffered so unconditionally succeed on BIO_flush().
-  case BIO_CTRL_FLUSH:
```
Comment (jablko) [inline iocore/net/BIO_fastopen.cc]: Now this case is delegated to the ordinary socket BIO:

```C
return BIO_meth_get_ctrl(const_cast<BIO_METHOD *>(BIO_s_socket()))(bio, cmd, larg, ptr);
```

Do I understand this feedback right?

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1328/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1222/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1330/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1224/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1347/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1240/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1381/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1274/ for details.

**Comment by zwoop** [comment]
You need this for 7.1.x ?

**Comment by zwoop** [comment]
Cherry-picked to 7.1.x

**Pull request merged**